Class decoupling

Chris-Pyles opened the pull request:
* Classes are now decoupled from workshop spreadsheet data.
+ Students who have duplicate preferences have the duplicates removed before the student object is constructed.
+ Students who have fewer than 6 preferences have the least popular workshops added to the end of their preference lists.

Closes #28

Chris-Pyles committed:
Decoupled classes from the spreadsheet data

Session class constructor now takes capacity
Workshop class constructor now takes name, number, and capacity
Student class constructor now takes firstName, lastName, and an array of preferences

Added global variable WORKSHOP_DICT - a dictionary of all workshop objects that can be indexed by workshop number

Students who have fewer than 6 preferences are now assigned extra preferences starting at the least popular workshop and incrementing upwards.

Chris-Pyles committed:
Fixed mistake where prettifying was undone

Chris-Pyles committed:
Fixed further mistakes where prettifying was undone

Chris-Pyles committed:
Fixed final issues with removing prettifying

beszel committed:
Fix linting errors

Also disabled some linting rules that don't make sense for this project.

beszel commented:
@Chris-Pyles I went ahead and resolved some lint errors. I also disabled some linting rules, like `prefer-const`, since Google Apps Script doesn't support the `const` keyword.
I'll take a closer look when I get the chance!

beszel commented on main.js:
```diff
@@ -176,75 +158,55 @@ var Workshop = function(row, workshopData, responseData) {
 
 /**
- * Returns an array of Workshop objects based on the workshop sheet and response sheets.
+ * Returns a dictionary of Workshop objects based on the workshop sheet and response sheet.
```
Even though we sometimes call this a dictionary, it's actually a JavaScript object.

beszel commented on main.js:
```diff
@@ -176,75 +158,55 @@ var Workshop = function(row, workshopData, responseData) {
     var workshopData = workshopSheet.getDataRange().getValues();
 
-    var workshopArray = [];
+    var workshopDict = {};
```
If JavaScript conventions are like any others, variable names don't need to say the type of the variable. Consider `workshops`. There are a few similar variables if you search for `array`.

beszel commented on main.js:
```diff
@@ -176,75 +158,55 @@ var Workshop = function(row, workshopData, responseData) {
-// An array containing every workshop object, sorted from least to most popular
-var WORKSHOP_ARRAY = makeWorkshopArray().sort(morePopular);
+// A dictionary containing all the workshops that can be indexed by workshop number
+var WORKSHOP_DICT = makeWorkshopDict();
```
Consider putting variables like this one at the top under the global constants, for consistency. The function definitions should be okay with it.

beszel reviewed (commented):
I thought I had more, but then I looked up some style stuff and it looks like capitalized globals are correct in JavaScript.

Chris-Pyles committed:
Replaced all instances of "let" with "var"

"let" is not supported by google apps script

Chris-Pyles committed:
Changed variable names to better ones

WORKSHOP_DICT is now NUMBERED_WORKSHOPS
WORKSHOP_ARRAY is now POPULAR_WORKSHOPS

beszel reviewed (approved):
*Good and cool*
